packages/curve-ui-kit/src/shared/ui/LargeTokenInput.tsx

```diff
@@ -261,7 +261,13 @@ export const LargeTokenInput = ({
   testId,
 }: Props) => {
   const [percentage, setPercentage] = useState<Decimal | undefined>(undefined)
-  const [balance, setBalance] = useDebounce(externalBalance, Duration.FormDebounce, onBalance)
+  const [balance, setBalance, cancelSetBalance] = useUniqueDebounce({
+    defaultValue: externalBalance,
+    callback: onBalance,
+    debounceMs: Duration.FormDebounce,
+    // We don't want to trigger onBalance if the value is effectively the same, e.g. "0.0" and "0.00"
+    equals: (a, b) => new BigNumber(a ?? 0).isEqualTo(b ?? 0),
```

Comment:
ser you just added this to the dependencies of a callback and it changes every render. Pls move it to the global scope